presto-iceberg/src/main/java/com/facebook/presto/iceberg/RefsTable.java

```diff
@@ -0,0 +1,107 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.predicate.TupleDomain;
+import com.facebook.presto.iceberg.util.PageListBuilder;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.ConnectorPageSource;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.FixedPageSource;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.SystemTable;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.google.common.collect.ImmutableList;
+import org.apache.iceberg.Table;
+
+import java.util.List;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class RefsTable
+        implements SystemTable
+{
+    private final ConnectorTableMetadata tableMetadata;
+    private final Table icebergTable;
+
+    private static final List<ColumnMetadata> COLUMNS = ImmutableList.<ColumnMetadata>builder()
+            .add(new ColumnMetadata("name", VARCHAR))
+            .add(new ColumnMetadata("type", VARCHAR))
+            .add(new ColumnMetadata("snapshot_id", BIGINT))
+            .add(new ColumnMetadata("max_reference_age_in_ms", BIGINT))
+            .add(new ColumnMetadata("min_snapshots_to_keep", BIGINT))
+            .add(new ColumnMetadata("max_snapshot_age_in_ms", BIGINT))
+            .build();
+
+    public RefsTable(SchemaTableName tableName, Table icebergTable)
+    {
+        tableMetadata = new ConnectorTableMetadata(requireNonNull(tableName, "tableName is null"), COLUMNS);
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        return new FixedPageSource(buildPages(tableMetadata, icebergTable));
+    }
+
+    private static boolean checkNonNull(Object object, PageListBuilder pagesBuilder)
```

Comment:
It seems to only be used for bigint values, so I'd just call this something like `appendLongValue` and append null if it's null, otherwise append the long value.

Reply:
Actaully I just noticed `value.minSnapshotsToKeep()` is an Integer. I can add it as `appendLongValue` itself and use `appendLongValue` method. Or I can keep this method like earlier and append it with `pagesBuilder.appendInteger` LMK?

Reply:
I'm just wondering if we can encapsulate the setting of values, either null or bigint, into one method.  So perhaps just `appendValue`?